archinstall/lib/interactions/disk_conf.py

```diff
@@ -366,20 +382,25 @@ def suggest_multi_disk_layout(
 	possible_devices = list(filter(lambda x: x.device_info.total_size >= min_home_partition_size, devices))
 	home_device = max(possible_devices, key=lambda d: d.device_info.total_size) if possible_devices else None
 
-	# find proper device for /root
-	devices_delta = {}
-	for device in devices:
-		if device is not home_device:
-			delta = device.device_info.total_size - desired_root_partition_size
-			devices_delta[device] = delta
-
-	sorted_delta: List[Tuple[disk.BDevice, Any]] = sorted(devices_delta.items(), key=lambda x: x[1])
-	root_device: Optional[disk.BDevice] = sorted_delta[0][0]
+	# find proper device for /root (size >= min_root_size and not home)
+	root_possible_devices: list[disk.BDevice] = [
+		device
+		for device in devices
+		if device.device_info.total_size >= min_root_partition_size
+		and device is not home_device
+	]
+
+	# we take the smallest disk >= min_root_size
+	root_device: Optional[disk.BDevice] = (
```

Comment:
This doesn't do what you think it is, it's not going to be a `Optional[disk.BDevice` but a `List[Optional[disk.BDevice]]`